internal/app/go-filecoin/internal/submodule/wallet_submodule.go

```diff
@@ -27,5 +30,6 @@ func NewWalletSubmodule(ctx context.Context, repo walletRepo) (WalletSubmodule,
 
 	return WalletSubmodule{
 		Wallet: fcWallet,
+		Signer: state.NewSigner(chain.ActorState, chain.ChainReader, fcWallet),
```

Comment:
I don't think a dependency from wallet to chain is acceptable. That resolution should happen higher level.

Reply:
I agree that wallet to chain is a bad dependency. This is the wallet submodule. It's already created after the chain submodule in the builder, and every submodule that depends on it also depends on the chain submodule. It could be renamed to something that more obviously depends on chain.